bids/analysis/tests/test_transformations.py

```diff
@@ -4,21 +4,38 @@
 from bids.variables.entities import RunInfo
 from bids.variables.kollekshuns import BIDSRunVariableCollection
 from bids.layout import BIDSLayout
+import math
 import pytest
 from os.path import join, sep
 from bids.tests import get_test_data_path
 import numpy as np
 import pandas as pd
 
 
+# Sub-select collection for faster testing, without sacrificing anything
+# in the tests
+SUBJECTS = ['01', '02']
+NRUNS = 3
+SCAN_LENGTH = 480
+
+cached_collections = {}
+
+
 @pytest.fixture
 def collection():
-    layout_path = join(get_test_data_path(), 'ds005')
-    layout = BIDSLayout(layout_path)
-    collection = layout.get_collections('run', types=['events'],
-                                        scan_length=480, merge=True,
-                                        sampling_rate=10)
-    return collection
+    if 'ds005' not in cached_collections:
+        layout_path = join(get_test_data_path(), 'ds005')
+        layout = BIDSLayout(layout_path)
+        cached_collections['ds005'] = layout.get_collections(
+            'run',
+            types=['events'],
+            scan_length=SCAN_LENGTH,
+            merge=True,
+            sampling_rate=10,
```

Comment:
btw, IIRC interesting "effects" did appear when I set it to 0.31, you get wonderful
```
_____________________________________________ test_resample_dense ______________________________________________

collection = <bids.variables.kollekshuns.BIDSRunVariableCollection object at 0x7fae82ce3810>

    def test_resample_dense(collection):
        new_sampling_rate = 50
        old_sampling_rate = collection.sampling_rate
        upsampling = float(new_sampling_rate) / old_sampling_rate
    
        collection['RT'] = collection['RT'].to_dense(old_sampling_rate)
        old_rt = collection['RT'].clone()
>       collection.resample(new_sampling_rate, in_place=True)

bids/analysis/tests/test_transformations.py:192: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
bids/variables/kollekshuns.py:279: in resample
    kind=kind)
bids/variables/variables.py:460: in resample
    f = interp1d(x, self.values.values.ravel(), kind=kind)
/usr/lib/python2.7/dist-packages/scipy/interpolate/interpolate.py:433: in __init__
    _Interpolator1D.__init__(self, x, y, axis=axis)
/usr/lib/python2.7/dist-packages/scipy/interpolate/polyint.py:60: in __init__
    self._set_yi(yi, xi=xi, axis=axis)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <scipy.interpolate.interpolate.interp1d object at 0x7fae81d4f050>
yi = array([0.000e+00, 1.793e+00, 1.637e+00, 0.000e+00, 0.000e+00, 0.000e+00,
     ...e+00, 0.000e+00,
       1.440e+00, 0.000e+00, 2.071e+00, 0.000e+00, 0.000e+00])
xi = array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  11,  12,
      ..., 880, 881, 882, 883,
       884, 885, 886, 887, 888, 889, 890, 891, 892, 893])
axis = -1

    def _set_yi(self, yi, xi=None, axis=None):
        if axis is None:
            axis = self._y_axis
        if axis is None:
            raise ValueError("no interpolation axis specified")
    
        yi = np.asarray(yi)
    
        shape = yi.shape
        if shape == ():
            shape = (1,)
        if xi is not None and shape[axis] != len(xi):
>           raise ValueError("x and y arrays must be equal in length along "
                             "interpolation axis.")
E           ValueError: x and y arrays must be equal in length along interpolation axis.
```

Reply:
Ah, good catch. Maybe open a separate issue for that.

Reply:
isn't it https://github.com/bids-standard/pybids/issues/361?

Reply:
It is!

Reply:
@yarikoptic great minds fail alike